File: front-end/src/types/TimeEntry.ts
export interface TimeEntry {
  ID: number;
  Description: string;
  ProjectID: number | null;
  StartDate: string;
  EndDate: string;
}

export interface TimeEntryCreate {
  Description: string;
  ProjectID: number | null;
  StartDate: string;
  EndDate: string;
}

export interface AssignProjectPayload {
  ProjectID: number | null;
}

File: back-end/models/TimeEntry.go
package models

import "time"

type TimeEntry struct {
	ID          int       `json:"ID"`
	Description string    `json:"Description"`
	ProjectID   *int      `json:"ProjectID"`
	StartDate   time.Time `json:"StartDate"`
	EndDate     time.Time `json:"EndDate"`
}

type TimeEntryCreate struct {
	Description string    `json:"Description"`
	ProjectID   *int      `json:"ProjectID"`
	StartDate   time.Time `json:"StartDate"`
	EndDate     time.Time `json:"EndDate"`
}

// swagger:model
type AssignProjectPayload struct {
	ProjectID *int `json:"ProjectID"`
}
